runtime/gc_structs/ClassFCCSlotIterator.hpp

```diff
@@ -0,0 +1,69 @@
+
+/*******************************************************************************
+ * Copyright (c) 1991, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup GC_Structs
+ */
+
+#if !defined(CLASSFCCSLOTITERATOR_HPP_)
+#define CLASSFCCSLOTITERATOR_HPP_
+
+#include "j9.h"
+#include "j9nonbuilder.h"
+
+/**
+ * Iterate through slots in the flattened class cache which may contain a
+ *  class reference.
+ *
+ * @see GC_ClassFCCSLotIterator
+ * @ingroup GC_Structs
+ */
+class GC_ClassFCCSlotIterator
+{
+private:
+    J9Class* _clazz;
+    UDATA _numberOfFlattenedFields;
+    UDATA _index;
+public:
+	GC_ClassFCCSlotIterator(J9Class *clazz) :
+        _clazz(clazz),
+        _numberOfFlattenedFields(0),
+        _index(0)
+	{
+        if ((J9_ARE_ALL_BITS_SET(_clazz->romClass->modifiers, J9AccValueType)) && (NULL != clazz->flattenedClassCache)) {
+            _numberOfFlattenedFields = clazz->flattenedClassCache->numberOfEntries;
```

Comment:
@dmitripivkine 
Does it matter if I use clazz or _clazz for standard?
Also is the` if((condition1)&&(condition2)) `fine?

Reply:
1) yes `clazz`(input parameter) and `_class`(private class member) are equivalent here, you can use any (but the same for consistency) 
2) yes, multiple logical conditions in one if is ok. `J9_ARE_ALL_BITS_SET` is a macro, so you don't need extra round brackets around it. The guide line is put extra round brackets in case there are some operations inside to not rely on order of operations.